Sandbox

nd-r opened the pull request:
review pls. thank you.

zwirec committed:
fix all

Commit:
pug?

Commit:
some changes

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

# Conflicts:
#	static/js/game/gameLogic/gameState.ts

8coon commented on src/js/game/Box2D/Box2D.ts:
```diff
@@ -0,0 +1,94 @@
+/*
```
Уберите это из проекта, пожалуйста.
Пакеты, которых нет в npm, можно ставить через `npm install git+https://git@github.com/somebody/someproject.git`

Commit:
added box2D dependency

8coon commented on src/js/application.ts:
```diff
@@ -0,0 +1,134 @@
+
+  /**
+   * Creates an instance of Application.
+   * @memberof Application
```
Думаю, что эта директива jsdoc тут лишняя. Вы же используете клёвые классы

8coon commented on src/js/blocks/block.ts:
```diff
@@ -0,0 +1,81 @@
+ * Base class for block implementations
+ *
+ * @export
+ * @class Block
```
То же самое про директиву jsdoc

8coon commented on src/js/blocks/input.js:
```diff
@@ -0,0 +1,68 @@
+   */
+  constructor(elem) {
+    super(elem);
+    console.log(elem);
```
console.log

8coon commented on src/js/blocks/input.js:
```diff
@@ -0,0 +1,68 @@
+export default class Input extends Block {
+  /**
+   * Builds the input block
+   * @param {HTMLElement} elem container of input and error placeholder
```
ts вперемешку с js, вы уж определитесь

8coon commented on src/js/blocks/input.js:
```diff
@@ -0,0 +1,68 @@
+   * Sets ok state
+   * @private
+   */
+  _setOKInputState() {
```
Если уж вы пишете на ts, то пишите тогда всё на ts. То есть этот метод должен быть `private setOKInputState`

8coon commented on src/js/blocks/loginForm.js:
```diff
@@ -0,0 +1,77 @@
+ * @class LoginForm
+ * @extends {Form}
+ */
+export default class LoginForm extends Form {
```
То же самое про ts/js

8coon commented on src/js/blocks/loginForm.js:
```diff
@@ -0,0 +1,77 @@
+        this._validatePassword(),
+      ];
+
+      errCodes = errCodes.filter((err) => err !== null);
```
Можно заменить это на some/every

8coon commented on src/js/blocks/mapTile.ts:
```diff
@@ -0,0 +1,45 @@
+import '../../img/img_placeholder.jpg';
+
+export default class MapTile {
+    constructor(mapMeta: Map.Meta) {
```
`constructor (private mapMeta: Map.Meta)` создаст приватное поле у класса и сразу присвоит туда значение из аргумента конструктора
[Подробности](https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/)

8coon commented on src/js/blocks/mapTile.ts:
```diff
@@ -0,0 +1,45 @@
+        const playersCount = document.createElement('p');
+        playersCount.innerHTML = `Players: ${this.mapMeta.players} / 4`;
+        playersCount.classList.add('main-frame__lobby-content__maps__map-tile__name');
+        this.htmlElement.appendChild(playersCount);
```
Это что, манипуляция DOM в императивном стиле????? :)

8coon commented on src/js/game/README.md:
```diff
@@ -0,0 +1,29 @@
+## Overview ##
```
Файлик от либы забыли удалить

8coon commented on src/js/game/board.ts:
```diff
@@ -0,0 +1,100 @@
+            //     this._canvas.renderAll.bind(this._canvas);
+            // });
+            // this._canvas.add(img);
+            const change = document.getElementById('img-width') as HTMLInputElement;
```
Зачем это?

8coon commented on src/js/game/board.ts:
```diff
@@ -0,0 +1,100 @@
+            });
+        });
+        this._canvas.on('object:moving', (option) => {
+            let s: boolean = false;
```
Отличное название для булевской переменной: s

8coon commented on src/js/game/board.ts:
```diff
@@ -0,0 +1,100 @@
+                    option.target.setPatternFill({ source: this.img.getElement() });
+                });
+            }
+            s = false;
```
А это зачем? Переменная-то локальная

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+        this.bodyDef.type = option.bodyType || b2BodyType.b2_staticBody;
+        this.bodyDef.userData = this;
+        this.fixDefs = fixDefs;
+        shape.originX = 'center';
```
Можно было бы и пустую строку вставить, а то стена текста

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+        shape.lockScalingX = shape.lockScalingY = option.lockScale || true;
+        this._shapes = shape;
+        this._shapes.set('objectCaching', false);
+        this._shapes.toObject = () => {
```
`() => ({id: this._ID})`
Но вы подмешиваете свой метод в библиотечный класс, это плохая идея

zwirec replied:
костыль, но по другому никак

8coon replied:
Почему? Можно же все просто создать где-нибудь функцию shapeToObject и вызывать её для всех шейпов

zwirec replied:
мне нужно к объекту библиотеки прицепить userData в виде инстанса моего класса

8coon replied:
Ну ок, тогда это оправдано

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+    }
+
+    get shapes(): fabric.Object {
+        return this._shapes;
```
В чём смысл этого геттера/сеттера?

zwirec replied:
просто делать поле public?

8coon replied:
Да

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+        this._shapes = value;
+    }
+
+    get isDeleted(): boolean {
```
То же самое про смысл геттера/сеттера

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+        return 'rect';
+    }
+
+    toJSON(): Object {
```
Стоило бы дать описание формата данных, который возвращает этот метод, через interface. Чтобы потом откуда угодно можно было посмотреть, какие поля тут в наличии.
И в указании типов в ts вместо сложных типов (String, Object) лучше использовать примитивы (string, object)

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+    constructor(position: b2Vec2 = b2Vec2.ZERO, radius: number = 50, option: Options = {}) {
+        let fixDef = new b2FixtureDef();
+        fixDef.shape = new b2CircleShape(radius * PIXEL_TO_METERS);
+        if (option.key) {
```
Тут тоже бы не помешала бы пустая строка

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+    private fixture: b2FixtureDef;
+
+    constructor(position: b2Vec2 = b2Vec2.ZERO, radius: number = 50, option: Options = {}) {
+        let fixDef = new b2FixtureDef();
```
const

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+        return 'circle';
+    }
+
+    toJSON(): Object {
```
То же самое про интерфейс

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+}
+
+
+export class BucketBody extends Body {
```
Как-то много сущностей на один файл
Это уже простыня на 500 строк, лучше разнести по разным файлам

zwirec replied:
каждый тип тела в разном файле?

8coon replied:
Да, так будет лучше

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+            stroke: null,
+        });
+
+        let right = new fabric.Rect({
```
Почему везде let, вы же нигде не переприсваиваете значение

zwirec replied:
справедливо

8coon commented on src/js/game/body.ts:
```diff
@@ -0,0 +1,527 @@
+        return 'bucket';
+    }
+
+    toJSON(): Object {
```
То же самое про интерфейс

8coon commented on src/js/game/box2d.d.ts:
```diff
@@ -0,0 +1,4182 @@
+declare module "Common/b2Settings" {
```
Забыли файлик либы

8coon commented on src/js/game/gameLogic/gameService.ts:
```diff
@@ -0,0 +1,260 @@
+
+    constructor(game: Game) {
+        this.game = game;
+        this.socket = new WebSocket('ws://194.87.110.17/backend/handler');
```
Хардкооод! Вынесите в конфиг, он же у вас есть

8coon commented on src/js/game/gameLogic/gameService.ts:
```diff
@@ -0,0 +1,260 @@
+    }
+
+    static createMap(canvases: HTMLCanvasElement | string): Board {
+        let canvas = new fabric.Canvas(canvases);
```
Ничего себе ...

zwirec replied:
что такое?

8coon replied:
Метод очень уж длинный

8coon commented on src/js/game/gameLogic/gameService.ts:
```diff
@@ -0,0 +1,260 @@
+        bodies.forEach((value, index, array) => {
+            mapBody.set(index, value);
+        });
+        return new Board(mapBody, canvas);
```
... вот это метод! 184-60=124 строки

8coon commented on src/js/game/gameLogic/gameService.ts:
```diff
@@ -0,0 +1,260 @@
+        let cvs = new fabric.Canvas(canvas);
+        let bodies: Map<number, Body> = new Map<number, Body>();
+        let json;
+        let resp;
```
Используйте const

8coon commented on src/js/game/gameLogic/gameService.ts:
```diff
@@ -0,0 +1,260 @@
+            });
+        }
+        catch (e) {
+            console.log('Network pizdec');
```
sooqa blyat'

8coon commented on src/js/game/gameLogic/gameState.ts:
```diff
@@ -0,0 +1,202 @@
+    }
+}
+
+export class PrepareState extends GameState {
```
Многовато сущностей на один файл

8coon commented on src/js/game/gameLogic/gameState.ts:
```diff
@@ -0,0 +1,202 @@
+        // }).then(resp => {
+        //     setTimeout(() => {
+        //     }, 1000);
+        // });
```
Уберите закоментированный код

8coon commented on src/js/game/gameLogic/gameState.ts:
```diff
@@ -0,0 +1,202 @@
+    onLoad(canvas: HTMLCanvasElement | string): Promise<any> {
+        this.game.changeState(new LoadState(this.game));
+        return this.game.load(canvas);
+        // throw new Error('Method not implemented.');
```
Ну вы же уже имплементировали метод! Зачем throw закомментировали-то? Могли бы спокойно удалить

8coon commented on src/js/game/index.ts:
```diff
@@ -0,0 +1,35 @@
+function init() {
+    let game = new Game({});
+
+    game.load('c').then();
```
А что случилось с game.load('a') и game.load('b')?
И в чём смысл этого then()

8coon commented on src/js/game/index.ts:
```diff
@@ -0,0 +1,35 @@
+    game.load('c').then();
+
+
+    document.getElementById('btn-sbscr').addEventListener('click', () => {
```
эт очнь удбнй и пнтнй слктр

8coon commented on src/js/modules/baseView.ts:
```diff
@@ -0,0 +1,86 @@
+ * Implements the base view class
+ *
+ * @export
+ * @class BaseView
```
Ненужная директива jsdoc

8coon commented on src/js/modules/router.ts:
```diff
@@ -0,0 +1,102 @@
+    this.currentView = info.view;
+  }
+
+  private goNotFound() {
```
Может быть routeNotFound? Или onRouteNotFound?

8coon commented on src/js/online.ts:
```diff
@@ -0,0 +1,59 @@
+  }
+
+  start(): void {
+    console.log(this.userService.getUser());
```
console.log

8coon commented on src/js/services/mapService.ts:
```diff
@@ -0,0 +1,44 @@
+    // public getMap(id: number): Map {
+    //     http.prGet(`${PATHS.GET_MAP}/${id}`)
+    //         .then((data) => this.loadedMap = data as Map)
+    //         .catch((err) => {
```
Закоментированный код

8coon commented on src/js/services/userService.ts:
```diff
@@ -0,0 +1,146 @@
+        throw UNEXPECTED_ERR;
+
+      default:
+        Utils.debugError('WTF?', response);
```
i dunno

nd-r replied:
![image](https://user-images.githubusercontent.com/22146930/33207702-bc78aae6-d11f-11e7-984b-fffbaa1fe9b0.png)

8coon commented on src/js/services/userService.ts:
```diff
@@ -0,0 +1,146 @@
+          throw response.body.message;
+        }
+
+        Utils.debugError('Something strange here');
```
Необъяснимо, но факт
==================
А если серьёзно, "WTF", "Something strange" -- как вы потом в консоли определите, что именно пошло не так? Ворнингам лучше быть хоть немного информативными

nd-r replied:
По идее до сюда не должны добраться(unreachable code) Но если вдруг случится, то надо выяснять что пошло не так

8coon replied:
Вполне может так случиться, что не будет response.body.message, и тогда вы провалитесь сюда. Это не unreachable code, и надо бы в дебаг вывести описание того, что произошло ("Incorrect response", например)

8coon commented on src/js/typings/globals.d.ts:
```diff
@@ -0,0 +1,76 @@
+  }
+}
+
+declare namespace Map {
```
Зачем namespace, если в нём один интерфейс? И нигде, кроме этого файла, вы неймспейсы не создаёте

nd-r replied:
Тут должны были быть body и другие объекты карты

8coon commented on src/js/views/loginView/loginView.pug:
```diff
@@ -0,0 +1,31 @@
+        .main-frame__content__content-column
+            .stub
+            form.main-frame__content__content-column__form
+                input.main-frame__content__content-column__form__username(type='text', name='username', placeholder='username', value='')
```
bem overdoze

Commit:
webpack fixes

Commit:
webpack fixes

Commit:
rm public.zip

Commit:
modified:   .gitignore
	deleted:    nodeServer/index.js
	deleted:    static/css/form-add-username.css
	deleted:    static/css/game-container.css
	deleted:    static/css/main.css
	deleted:    static/fonts/CabinSketch.woff2
	deleted:    static/fonts/CabinSketchBold.woff2
	deleted:    static/fonts/KGSecondChancesSketch.ttf
	deleted:    static/icons/facebook-logo.svg
	deleted:    static/icons/pointer-arrow-sketch-white.gif
	deleted:    static/icons/vkontakte-draw.svg
	deleted:    static/img/answer-box.png
	deleted:    static/img/border-chalk.png
	deleted:    static/img/chalk-background-filled.png
	deleted:    static/img/chalk_border_big.png
	deleted:    static/img/chalk_input_border.png
	deleted:    static/img/chalk_input_border_error.png
	deleted:    static/img/chalkboard.jpg
	deleted:    static/img/options_icon.svg
	deleted:    static/index.html
	deleted:    static/js/Offline/offline.ts
	deleted:    static/js/Offline/views/lobbyView/lobbyView.css
	deleted:    static/js/Offline/views/lobbyView/lobbyView.pug
	deleted:    static/js/Offline/views/lobbyView/lobbyView.ts
	deleted:    static/js/Online/online.ts
	deleted:    static/js/Online/views/lobbyView/lobbyView.css
	deleted:    static/js/Online/views/lobbyView/lobbyView.pug
	deleted:    static/js/Online/views/lobbyView/lobbyView.ts
	deleted:    static/js/Online/views/loginView/loginView.css
	deleted:    static/js/Online/views/loginView/loginView.pug
	deleted:    static/js/Online/views/loginView/loginView.ts
	deleted:    static/js/Online/views/signUpView/signUpView.css
	deleted:    static/js/Online/views/signUpView/signUpView.pug
	deleted:    static/js/Online/views/signUpView/signUpView.ts
	deleted:    static/js/application.ts
	deleted:    static/js/blocks/block.ts
	deleted:    static/js/blocks/button.js
	deleted:    static/js/blocks/form.js
	deleted:    static/js/blocks/input.js
	deleted:    static/js/blocks/loginForm.js
	deleted:    static/js/blocks/signUpForm.js
	deleted:    static/js/config/config.ts
	deleted:    static/js/config/paths.js
	deleted:    static/js/game/Box2D/Box2D.js
	deleted:    static/js/game/Box2D/Box2D.ts
	deleted:    static/js/game/Box2D/Collision/Shapes/b2ChainShape.js
	deleted:    static/js/game/Box2D/Collision/Shapes/b2ChainShape.ts

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

# Conflicts:
#	static/js/game/body.ts
#	static/js/game/gameLogic/gameService.ts
#	static/js/views/gameView/gameView.ts

zwirec committed:
Fix game part

zwirec committed:
Fix game part
